mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -164,9 +164,40 @@ impl TendermintConf {
     }
 }
 
+struct TendermintRpcClient(AsyncMutex<TendermintRpcClientImpl>);
+
+struct TendermintRpcClientImpl {
+    rpc_clients: Vec<HttpClient>,
+}
+
+#[async_trait]
+impl RpcCommonOps for TendermintCoin {
+    type RpcClient = HttpClient;
+    type Error = TendermintCoinRpcError;
+
+    async fn get_live_client(&self) -> Result<Self::RpcClient, Self::Error> {
+        let mut client_impl = self.client.0.lock().await;
+        // try to find first live client
+        for (i, client) in client_impl.rpc_clients.clone().into_iter().enumerate() {
```

Comment:
Can we iterate the RPC clients by reference?
```rust
for (i, client) in client_impl.rpc_clients.iter().enumerate() {}
```

Reply:
No, we cant, bcz in this case we have immutable and mutable borrowing in one time.
```
error[E0502]: cannot borrow `client_impl` as mutable because it is also borrowed as immutable
   --> mm2src/coins/tendermint/tendermint_coin.rs:189:17
    |
181 |         for (i, client) in client_impl.rpc_clients.iter().enumerate() {
    |                            ----------- immutable borrow occurs here
...
189 |                 client_impl.rpc_clients.rotate_left(i);
    |                 ^^^^^^^^^^^ mutable borrow occurs here
190 |                 return Ok(client.clone());
    |                           -------------- immutable borrow later used here
```
So, we need to clone vector of clients.